src/plotly.js

```diff
@@ -33,14 +33,23 @@ exports.MathJaxConfig = require('./fonts/mathjax_config');
 exports.defaultConfig = require('./plot_api/plot_config');
 
 // plots
-exports.Plots = require('./plots/plots');
+var Plots = exports.Plots = require('./plots/plots');
+
+var Cartesian = require('./plots/cartesian');
+Plots.registerSubplot(Cartesian);
+
 exports.Axes = require('./plots/cartesian/axes');
 exports.Fx = require('./plots/cartesian/graph_interact');
-exports.Scene = require('./plots/gl3d/scene');
-exports.Gl3dLayout = require('./plots/gl3d/layout');
-exports.Geo = require('./plots/geo/geo');
-exports.GeoLayout = require('./plots/geo/layout');
-exports.Scene2D = require('./plots/gl2d/scene2d');
+
+var Geo = require('./plots/geo');
+Plots.registerSubplot(Geo);
```

Comment:
one step towards modularity. This register step will taken out of the *core* module and added on-demand by users.